### PR TITLE
Solution, week 1 exercise

### PR DESCRIPTION
**Pull request by shreyanjoshi**

### CODE REVIEW

File: exercises/week 1/solutions/shreyan.sh [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+else
+   echo "Failed to download Bitcoin Core binary."
+   exit 1
+fi
+
+
+
+
+# Set the data directory for the Bitcoin Core
+bitcoin_data_dir="/home/shreyan/.bitcoin/"
```
Comment (rajarshimaitra) [inline exercises/week 1/solutions/shreyan.sh:63]: should be `~/.bitcoin` to make it system agnostic.

File: exercises/week 1/solutions/shreyan.sh [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+mkdir -p "$bitcoin_data_dir"
+
+# Create and populate the bitcoin.conf file
+echo "regtest=1" > "$bitcoin_data_dir/bitcoin.conf"
+echo "fallbackfee=0.0001" >> "$bitcoin_data_dir/bitcoin.conf"
+echo "server=1" >> "$bitcoin_data_dir/bitcoin.conf"
+echo "txindex=1" >> "$bitcoin_data_dir/bitcoin.conf"
+
+# Start bitcoind
+bitcoind -regtest -daemon
```
Comment (rajarshimaitra) [inline exercises/week 1/solutions/shreyan.sh:75]: After setting server=1 in conf file `-daemon` isn't required anymore.

File: exercises/week 1/solutions/shreyan.sh [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+
+# Set the data directory for the Bitcoin Core
+bitcoin_data_dir="/home/shreyan/.bitcoin/"
+
+# Create the data directory if it doesn't exist
+mkdir -p "$bitcoin_data_dir"
+
+# Create and populate the bitcoin.conf file
+echo "regtest=1" > "$bitcoin_data_dir/bitcoin.conf"
```
Comment (rajarshimaitra) [inline exercises/week 1/solutions/shreyan.sh:69]:
```suggestion
echo "regtest=1" >> "$bitcoin_data_dir/bitcoin.conf"
```

File: exercises/week 1/solutions/shreyan.sh [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+bitcoind -regtest -daemon
+
+# Wait for bitcoind to start
+sleep 2
+
+# Create the Miner Wallet
+bitcoin-cli -rpcwallet= createwallet Miner
+
+# Create the Trader Wallet
+bitcoin-cli -rpcwallet= createwallet Trader
```
Comment (rajarshimaitra) [inline exercises/week 1/solutions/shreyan.sh:84]: no need for `-rpcwallet` here.

File: exercises/week 1/solutions/shreyan.sh [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+# Generate an address for Miner wallet with the label “Mining Reward”
+mining_reward_address=$(bitcoin-cli -rpcwallet=Miner getnewaddress "Mining Reward")
+echo "Mining Reward Address: $mining_reward_address"
+
+# Mine Blocks
+blocks_to_mine=100
```
Comment (rajarshimaitra) [inline exercises/week 1/solutions/shreyan.sh:91]: 1 more block before you get spendable balance.

**Review by rajarshimaitra — COMMENTED** [review]
The script needs some changes to run in generic ways. Plus few nits.

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
https://github.com/Bitshala/LBTCL-Cohort/pull/6#pullrequestreview-1552007436

**Comment by shreyanjoshi** [comment]
Making a separate pull request with generic script, well-defined functions, and using a specific utxo to make the transaction to output change address and amount. Apologies for the inconvenience.

**Pull request closed without merging**